core/blockchain.go

```diff
@@ -544,7 +545,7 @@ func (bc *BlockChain) getJustifiedNumber(header *types.Header) uint64 {
 // getFinalizedNumber returns the highest finalized number before the specific block.
 func (bc *BlockChain) getFinalizedNumber(header *types.Header) uint64 {
 	if p, ok := bc.engine.(consensus.PoSA); ok {
-		return p.GetFinalizedNumber(bc, header)
+		return p.GetFinalizedHeader(bc, header).Number.Uint64()
```

Comment:
If we return the pointer, we should handle it properly, the pointer may be nil, and then this function will panic.